### ISSUE TITLE
[Feature request] Progress bar

### ISSUE THREAD

**Issue by eivindhammers** [jannes-m/RQGIS#57]
Would it make sense to add a progress bar to `run_qgis()`? I think it's possible through pyqgis, but no idea how.

**Comment by tim-salabim** [comment]
If ti is possible, it would be nice to have indeed!

**Comment by jannes-m** [comment]
Currently, we are rewriting RQGIS completely using the reticulate-package. Additionally, we would like to make RQGIS more user-friendly (see rgrass7::doGRASS). Once this is done, we can have a look into the progress bar.

**Comment by pat-s** [comment]
The problem which I see here is that most of the progress bar implementations I know of work on number of iterations in a loop or similar. 

The implementation in a loop example using `pyqgis` is explained in https://gis.stackexchange.com/questions/110524/qgis-processing-progress and http://docs.qgis.org/2.2/pdf/en/QGIS-2.2-PyQGISDeveloperCookbook-en.pdf in section 11.2

The problem that we face here is that we provide a generic interface to access all existing functions using the `processing` module of QGIS and do not just provide a plugin with one function. Right now I think we would then need to create a progress implementation for each function since every function is different in the way it does the processing (some work on pixels, some on vectors, etc.)? This is of course neither possible nor useful. 

On the other side, this https://gis.stackexchange.com/questions/235031/pyqgis-is-there-a-way-to-disable-the-progress-bar-in-the-processing-tool questions shows that there is already a progress bar implementation in `processing.runalg()` (which is the function we are using to run all algs). So we should look a bit deeper into this...

**Comment by jannes-m** [comment]
I think it is obvious that we had to use a QGIS message bar. How would R know the progress already made for a specific QGIS geoalgorithm. The problem here is, that the progress bar for a geoalgorithm depends on the QgisInterface (iface):

https://github.com/qgis/QGIS/blob/b3e1aba1cdc7b9398ceab0443984736c0cebd6ea/python/plugins/processing/core/Processing.py#L228-L229

However, iface is not available for external applications. Maybe, one can circumvent this (but I am not sure how). But we can ask the QGIS developers or https://gis.stackexchange.com/ for any hints.
